### PR TITLE
:bug: Source Github: return AirbyteMessage if max retry exeeded for 202 status code

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
resolved: https://github.com/airbytehq/oncall/issues/3294
They are receiving the Unexpected type for data_or_message: <class 'NoneType'>: None error.

## How
NoneType error occurs because we have incorrect handling of 202 status code response if max retry reached.  Added `yield AirbyteMessage` instead of `yield` to fix  NoneType error. So now we can skip this stream if data is still computing on github side after 6 retries.

### CODE REVIEW

**Review by maxi297 — APPROVED** [review]
🚢

File: airbyte-integrations/connectors/source-github/unit_tests/test_stream.py [hunk 1]
```diff
@@ -1381,9 +1381,9 @@ def test_stream_contributor_activity_accepted_response(caplog):
         status=202,
     )
     with patch("time.sleep", return_value=0):
-        list(read_full_refresh(stream))
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-github/unit_tests/test_stream.py:1384]: let's reproduce this bug in a test so we would not miss it again

**Review by davydov-d — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-github/unit_tests/test_stream.py (hunk 1 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-github/unit_tests/test_stream.py:1384]: this test reproduces this "bug", None is not returning now from read records so read results now can be correctly parsed in https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/utils/record_helper.py#L14.

**Review by davydov-d — APPROVED** [review]

**Comment by octavia-approvington** [comment]
Lets merge it <br/> sheep thumbs up

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**